src/common/eth-providers/consensus-provider/consensus-provider.service.ts

```diff
@@ -27,7 +27,7 @@ interface RequestRetryOptions {
   maxRetries?: number;
   dataOnly?: boolean;
   useFallbackOnRejected?: (e: any) => boolean;
-  useFallbackOnResolved?: (r: any) => boolean;
+  useFallbackOnResolved?: (r: any) => Promise<boolean>;
```

Comment:
looks unconsistent with `useFallbackOnRejected` a bit

Reply:
currently no place using `useFallbackOnRejected` and I don't see any reason to implement async on it yet. Maybe there is sense to remove it, but not now, I guess